### PR TITLE
new NSE table

### PR DESCRIPTION
**Pull request by zingale**
A new NSE table for aprox19 created with pynucastro replaces the original NSE table. 
This has mostly the same range of validity, but has finer dlogT spacing and also tabulates
dAbar/dt, d<B/A>/dt, and e_nu.

This also contains a notebook that can be used to inspect the true NSE result from pynucastro at an arbitrary thermodynamic point.

Finally, we include the python script that generates the table.

### CODE REVIEW

**Comment by zingale** [comment]
here's the test suite run that uses the table:
http://groot.astro.sunysb.edu/Microphysics/test-suite/gfortran/2023-11-06-001/test_react_C-aprox19-NSE-VODE.html

there are diffs, as should be expected.

File: nse_tabular/README.md [hunk 1]
```diff
@@ -1,150 +1,94 @@
...
+  Sometimes the solver can fail to converge, but may do so if a better
+  initial guess for the chemical potentials is provided.  There is an
+  attempt to cache the chemical potentials that worked for the last
+  temperature to hopefully accelerate the convergence.
+
+The script will take a long time to run.  Upon completion, the
+following should be copied into the on-grid network's subdirectory
+(e.g. `networks/aprox19/`):
+
+* `nse.tbl` : this is the table itself.  *It should really be renamed
```
[the remark this answers is not in the document]
Comment (zingale) [reply nse_tabular/README.md:82]: well, this is the output of the script and then it is suggested that it be renamed.  My goal is to make this script eventually support any net.

File: nse_tabular/make_nse_table.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    mu_n0 = -15.0
+
+    mu_p = np.ones((len(rhos), len(yes)), dtype=np.float64) * mu_p0
+    mu_n = np.ones((len(rhos), len(yes)), dtype=np.float64) * mu_n0
+
+    nse_net.generate_table(rho_values=rhos,
+                           T_values=Ts,
+                           Ye_values=yes,
+                           comp_reduction_func=get_aprox19_comp,
+                           verbose=True)
```
[the remark this answers is not in the document]
Comment (zingale) [reply nse_tabular/make_nse_table.py:150]: my goal is to eventually make it support any net.  I will make it clear in the README how to switch it to a different net.

File: nse_tabular/nse_table_data.cpp [hunk 3]
```diff
@@ -3,9 +3,12 @@
 #ifdef NSE_TABLE
 namespace nse_table
 {
-  AMREX_GPU_MANAGED amrex::Array1D<amrex::Real, 1, npts> abartab;
-  AMREX_GPU_MANAGED amrex::Array1D<amrex::Real, 1, npts> ebtab;
-  AMREX_GPU_MANAGED amrex::Array1D<amrex::Real, 1, npts> wratetab;
+    AMREX_GPU_MANAGED amrex::Array1D<amrex::Real, 1, npts> abartab;
```
[the remark this answers is not in the document]
Comment (zingale) [reply nse_tabular/nse_table_data.cpp:6]: fixed

**Review by zhichen3 — APPROVED** [review]

**Pull request merged**